### PR TITLE
feat(agent): Add config check sub-command

### LINKED ISSUE #15726: feat: flag to validate config files
### Use Case

My telegraf instance receives new config files automatically from an external service, so I'd like to validate the config file works before running it.  It should catch these errors:
- Invalid TOML
- Plugins which are not compiled into the binary
- Incorrect plugin options

This could be implemented as a new command-line flag `--check-config`.  If it passes, it returns 0, and if it fails, it returns 78 (or any other code except 1) to differentiate this from a failed `--test`.

The `--test` flag is inadequate for this purpose for 3 reasons:
- It returns 1 if any of the inputs fail, even though the config is valid
- It takes a long time because it has to wait for each input to timeout
- It doesn't work in a dev environment that doesn't have network access to all the targets

### Expected behavior

..

### Actual behavior

..

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by srebhan**
## Summary

This PR adds a `telegraf config check` sub-command to check if the configuration is valid and can be successfully parsed. All plugins are initialized to validate options but are **NOT** started or executed. The sub-command will exit with a non-zero error code if an error occurs and with a zero error-code otherwise.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #15726

### CODE REVIEW

File: cmd/telegraf/cmd_config.go [hunk 1]
```diff
@@ -32,6 +33,60 @@ func getConfigCommands(pluginFilterFlags []cli.Flag, outputBuffer io.Writer) []*
...
+		The 'check' command reads the configuration files specified via '--config' or
+		'--config-directory' and tries to initialize, but not start, the plugins.
+		Syntax and semantic errors detectable without starting the plugins will
+		be reported.
+		If no configuration file is	explicitly specified the command reads the
+		default locations and uses those configuration files.
+
+		To check the file 'mysettings.conf' use
+
+		> telegraf --config mysettings.conf config check
```
Comment (powersj) [inline cmd/telegraf/cmd_config.go:49]: This looks like a backwards way of doing this. I would have expected:

> telegraf config check --config...

I know you are trying to re-use the global config/config-dir variables, but is there a way to have these so they are local to the check command?

**Review by powersj — COMMENTED** [review]
Thanks for putting this together so quickly, I do have one comment/question

**Review by powersj — APPROVED** [review]
Thank you very much for taking a look at the flags!

**Review by DStrand1 — APPROVED** [review]

**Comment by srebhan** [comment]
@Hipska the issue with the previous documentation was that it was incorrect and a pain to maintain. If you do have a constructive proposal on how to document this with correctness and maintainability in mind, let me know!

**Pull request merged**